systemd/systemd.go

```diff
@@ -692,12 +705,56 @@ func IsTimeout(err error) bool {
 	return isTimeout
 }
 
+func (l Log) parseLogRawMessageString(key string, sliceHandler func([]string) (string, error)) (string, error) {
+	valObject, ok := l[key]
+	if !ok || valObject == nil {
+		// NOTE: journalctl says that sometimes if a json string would be too
+		// large null is returned, so we may miss a message here
+		return "", fmt.Errorf("key %q missing from message", key)
+	}
+
+	// first try normal string
+	s := ""
+	err := json.Unmarshal(*valObject, &s)
+	if err == nil {
+		return s, nil
+	}
+
+	// try runes next, this is the case if journald thinks the output is not
+	// safe ascii
+	r := []rune{}
+	err = json.Unmarshal(*valObject, &r)
+	if err == nil {
+		return string(r), nil
+	}
+
+	// finally try list of strings
+	stringSlice := []string{}
+	err = json.Unmarshal(*valObject, &stringSlice)
+	if err == nil {
+		// if the slice is of length 1, just promote it to a plain scalar string
+		if len(stringSlice) == 1 {
```

Comment:
I'm uncertain this transformation is safe. Can you please justify it?

Reply:
if we get json like:

```json
{
  "MESSAGE": ["hello world"]
}
```

this code just bubbles it up as if it was actually
```json
{
  "MESSAGE": "hello world"
}

```